download_and_create_reference_datasets/v02/hail_scripts/write_1kg_ht.py

```diff
@@ -46,14 +46,22 @@ def vcf_to_mt(path, genome_version):
 
     return all_mt
 
+def annotate_mt(mt):
+    # Annotate POPMAX_AF, which is max of respective fields using a_index for multi-allelics.
+    return mt.annotate_rows(POPMAX_AF=hl.max(mt.info.AFR_AF[mt.a_index-1],
```

Comment:
This is only works if the mt is bi-allelic. (A multiallelic mt would silently overwrite popmax for all except 1 of the alelles). Would be great to check/throw an error on multi-allelic mt.

Reply:
Multi-allelic variants are split in `vcf_to_mt` before this is run. The `a_index` field is added by [split_multi](https://hail.is/docs/0.2/methods/genetics.html#hail.methods.split_multi).

Reply:
Even though it's correct within the current pipeline, it's a hard to catch bug if the code is reused or upstream code is modified. Having methods that double check args adds overhead but I'd vote for making the assumption explicit in this case.

Reply:
It wouldn't be an obvious error message, but this should fail on non-split tables since the `a_index` field would not exist.

Reply:
Yeah I think it's important to remember the frequency with which this is run when talking about exception cases. In keeping with the spirit of this repo, a lot of cases are not considered haha. When I'm rewriting, I'm paying more attention to overall maintainability and cleanliness, but also the frequency with which it is used to allocate how much time is spent making it robust.

Reply:
I'm thinking of cases where a_index is added without splitting (like https://github.com/macarthur-lab/hail-elasticsearch-pipelines/blob/b8b2c7c5b50697941fe9737c7e12c592b0a74341/download_and_create_reference_datasets/v02/hail_scripts/write_1kg_ht.py#L36) - which was also necessary in v01 and led to this bug happening. 
I agree, it's quite unlikely to happen.

Reply:
Hmm, maybe I'm not understanding. Can you clarify in what cases it would report an incorrect info and what the incorrect info would be, as well as the check that would catch this?

Reply:
The current code is correct.. but if the upstream code ever broke in a way that ran this line:
` biallelic_mt = biallelic_mt.annotate_rows(a_index=1, was_split=False) `
on multiallelic variants, annotate_mt(mt) would become incorrect.
I guess this is just a reminder to watch out if/when modifying the splitting logic, since 
` biallelic_mt.annotate_rows(a_index=1, was_split=False) `
is still necessary in places. 
Checking for it explicitly would require filtering for  `hl.len(mt.alleles) > 2` and making sure the result is 0.